packages/rich-text/src/plugins/Hyperlink/components/EntityHyperlink.tsx

```diff
@@ -33,45 +35,45 @@ export type HyperlinkElementProps = {
 export function EntityHyperlink(props: HyperlinkElementProps) {
   const editor = useContentfulEditor();
   const sdk: FieldAppSDK = useSdkContext();
+  const focus = editor.selection?.focus;
   const { target } = props.element.data;
   const { onEntityFetchComplete } = useLinkTracking();
+  const pathToElement = findNodePath(editor, props.element);
+  const isLinkFocused = pathToElement && focus && Path.isChild(focus.path, pathToElement);
 
   const tooltipContent = useEntityInfo({
     target,
     sdk,
     onEntityFetchComplete,
   });
 
-  if (!target) return null;
-
-  function handleClick(event: React.MouseEvent<HTMLAnchorElement>) {
-    event.preventDefault();
-    event.stopPropagation();
-    if (!editor) return;
-
-    const p = fromDOMPoint(editor, [event.target as Node, 0]);
-
-    if (p) {
-      addOrEditLink(editor, sdk, editor.tracking.onViewportAction, p.path);
-    }
+  if (!target) {
+    return null;
   }
 
+  const popoverText = (
+    <Text fontColor="blue600" fontWeight="fontWeightMedium" className={styles.openLink}>
+      {tooltipContent}
+    </Text>
+  );
+
   return (
-    <Tooltip
-      content={tooltipContent}
-      targetWrapperClassName={styles.hyperlinkWrapper}
-      placement="bottom"
-      maxWidth="auto"
+    <LinkPopover
+      isLinkFocused={isLinkFocused}
+      handleEditLink={() => handleEditLink(editor, sdk, pathToElement)}
+      handleRemoveLink={() => handleRemoveLink(editor)}
+      popoverText={popoverText}
     >
-      <TextLink
-        as="a"
-        onClick={handleClick}
+      <Text
```

Comment:
Can't we use `TextLink` with `as="span"`? Then we would not need to re-implement the styles

Reply:
No you can only set it as `span` or `button`